sdk/core/azure-core/azure/core/serialization.py

```diff
@@ -95,29 +107,16 @@ class AzureJSONEncoder(JSONEncoder):
     """A JSON encoder that's capable of serializing datetime objects and bytes."""
 
     def default(self, o):  # pylint: disable=too-many-return-statements
+        if isinstance(o, (bytes, bytearray)):
```

Comment:
I think that `try`ing to return `super(AzureJSONEncoder, self).default(o)` first, and then trying the rest in the `except` block, is probably the most Pythonic approach.

You raise a good point though -- rather than calling `super(...).default(o)` both at the start and at the end, we can probably just raise the exception we got from the first attempt at the end of the method, since we know what will happen. So something like
```python
try:
    return super(AzureJSONEncoder, self).default(o)
except TypeError as e:
    (try different serializations)
    raise e
```